docs/docs.trychroma.com/markdoc/content/integrations/embedding-models/jina-ai.md

```diff
@@ -45,3 +45,40 @@ const collectionGet = await client.getCollection({name:"name", embeddingFunction
 {% /TabbedCodeBlock %}
 
 You can pass in an optional `model_name` argument, which lets you choose which Jina model to use. By default, Chroma uses `jina-embedding-v2-base-en`.
+
+### Late Chunking Example
+
+JinaAI supports late chunking on `jina-embeddings-v3`, which can enrich the context across the documents being inserted by combining multiple chunks.
+
+{% tabs group="code-lang" hideTabs=true %}
+{% Tab label="python" %}
+
+```python
+from chromadb.utils.embedding_functions import JinaEmbeddingFunction
+jinaai_ef = JinaEmbeddingFunction(
+                api_key="YOUR_API_KEY",
+                model_name="jina-embeddings-v3",
+                late_chunking=True,
+            )
```

Comment:
note, when model being set to jina-embeddings-v3, user is expected to set a task to get optimal performance.

in this example (QA/Retrieval) at indexing time `task=retrieval.passage`, at searching time, `task=retrieval.query` would be best.

however this might leads to 2 separate JinaEmbeddingFunction instances.. not very elegant, another option is set `task=text-matching`, this should also give better result than not offering a task. We need to think about how to handle it better

Reply:
the same is expected for the jupyter notebook below

Reply:
sounds good, i think other embedding providers also do the `task` trick, e.g. for different task use different encode function, i was wondering how Choma is handling that now?